language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/JoinClauseNodeContext.java

```diff
@@ -115,10 +129,65 @@ private boolean onSuggestBindingPattern(BallerinaCompletionContext context, Join
     private boolean onSuggestInKeyword(BallerinaCompletionContext context, JoinClauseNode node) {
         int cursor = context.getCursorPositionInTree();
         TypedBindingPatternNode typedBindingPattern = node.typedBindingPattern();
-        ExpressionNode expression = node.expression();
 
-        return node.inKeyword().isMissing() && !typedBindingPattern.isMissing()
-                && cursor > typedBindingPattern.textRange().endOffset()
-                && (expression.isMissing() || cursor < expression.textRange().startOffset());
+        if (!node.inKeyword().isMissing() || typedBindingPattern.isMissing()) {
+            return false;
+        } else if (cursor > typedBindingPattern.textRange().endOffset() &&
+                cursor <= node.expression().textRange().startOffset()) {
+            /*
+             * Captures:
+             * (1) join var varName <cursor>
+             */
+            return true;
+        } else {
+            Node nodeAtCursor = context.getNodeAtCursor();
+
+            /*
+             * Captures:
+             * (1) join var varName i<cursor>
+             * (2) join var varName i<cursor> expression
+             */
+            if (nodeAtCursor.kind() == SyntaxKind.SIMPLE_NAME_REFERENCE) {
+                SimpleNameReferenceNode nameReferenceNode = (SimpleNameReferenceNode) nodeAtCursor;
+                return node.expression().textRange().startOffset() + 1 == cursor &&
```

Comment:
Say, when at the cursor the particular letter typed is `n`. Then this logic is invalidated right? If so, I think we should drop the starts with check here.

Reply:
I have a question regarding your suggestion. If the user types `n` (anything other than `i`), IMO `in` is not a suggestion. Even if we return the completion item `in`, VSCode will filter that out because the letter already typed is starting from `n`. WDYT?

Reply:
I think, in this case, still the `in` keyword will be shown. Since the editor do the distance based filtering it will suggest the nearest matching entry.

Reply:
I just tested that. Seems like returning `in` while the user has types `o` doesn't show `in` as a suggestion in VSCode.

Reply:
Anyway the `startsWith` check is redundant as we check if the cursor is at the end of the expression. Therefore, removed that.